CarsTest.cpp

```diff
@@ -0,0 +1,87 @@
+#include <gtest/gtest.h>
+#include <memory>
+#include "PetrolEngine.hpp"
+#include "PetrolCar.hpp"
+#include "InvalidGear.hpp"
+#include "ElectricEngine.hpp"
+#include "ElectricCar.hpp"
+#include "HybridCar.hpp"
+
+// PETROL CAR //
+
+TEST(CarsTests, ShouldChangeGearThrowInvalidGear)
+{
+  PetrolCar opel(std::make_unique<PetrolEngine>(PetrolEngine(120, 1800, 6)));
```

Comment:
This is a common object that is used in majority of tests. You can extract it to a test fixture class and use TEST_F - https://google.github.io/googletest/primer.html#same-data-multiple-tests